SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -135,6 +136,9 @@ void USpatialReceiver::OnAddComponent(const Worker_AddComponentOp& Op)
 	case SpatialConstants::STARTUP_ACTOR_MANAGER_COMPONENT_ID:
 		GlobalStateManager->ApplyStartupActorManagerData(Op.data);
 		return;
+	case SpatialConstants::TOMBSTONE_COMPONENT_ID:
+		RemoveActor(Op.entity_id);
```

Comment:
The ordering here seems a little off. If we're processing this during a critical section (when initially checking out the entity), won't we attempt to receive the actor after removing it? I'm wondering if there should be some logic in `USpatialReceiver::TryGetOrCreateActor` that explicitly checks for this.

Reply:
updated ReceiveActor per our slack chat

Reply:
hmm tested on cloud deploy and the functionality doesn't work with this change (doesn't delete actor on 2..N joining clients).  Will investigate.

Reply:
fixed

Reply:
Worth noting in the worker wrapper you can query what tomestone components exist and then ignore the entities added associated with them.

Reply:
that's excellent :)